src/main/java/edu/ksu/canvas/attendance/submitter/AssignmentSubmitter.java

```diff
@@ -171,23 +192,48 @@ private String getCommentHeader(AttendanceSummaryModel.Entry entry, AttendanceCo
-        String absentPoints = attendanceAssignment.getAbsentPoints();
+    private double calculateStudentGrade(AttendanceAssignment attendanceAssignment, List<AttendanceSummaryModel> summaryForSections, boolean isSimpleAttendance, AttendanceStudent student) {
 
         if (isSimpleAttendance) {
-            int totalClasses = getTotalSimpleClasses(entry);
-            totalClasses = totalClasses == 0? 1 : totalClasses;
-            return ((((entry.getTotalClassesPresent() * (calculation(presentPoints))) +
-                    (entry.getTotalClassesTardy() * (calculation(tardyPoints))) +
-                    (entry.getTotalClassesExcused() * (calculation(excusedPoints))) +
-                    (entry.getTotalClassesMissed() * (calculation(absentPoints)))) / totalClasses) * Double.parseDouble(attendanceAssignment.getAssignmentPoints()));
+
+            int totalClasses = 0;
+            double presentPoints = calculation(attendanceAssignment.getPresentPoints());
+            double tardyPoints = calculation(attendanceAssignment.getTardyPoints());
+            double excusedPoints = calculation(attendanceAssignment.getExcusedPoints());
+            double absentPoints = calculation(attendanceAssignment.getAbsentPoints());
+            double totalClassesPresent = 0.0, totalClassesTardy = 0.0, totalClassesExcused = 0.0, totalClassesMissed = 0.0;
+
+            for (AttendanceSummaryModel model : summaryForSections) {
+                for (AttendanceSummaryModel.Entry entry : model.getEntries()) {
+                    if (entry.getSisUserId() != null && entry.getSisUserId().equals(student.getSisUserId())) {
+                        totalClasses += getTotalSimpleClasses(entry);
+                        totalClassesPresent += entry.getTotalClassesPresent();
+                        totalClassesTardy += entry.getTotalClassesTardy();
+                        totalClassesExcused += entry.getTotalClassesExcused();
+                        totalClassesMissed += entry.getTotalClassesMissed();
+                    }
+                }
+            }
+            if (totalClasses == 0){
+                return 0.0;
+            }
+            else {
+                return ((((totalClassesPresent * presentPoints) +
+                    (totalClassesTardy * tardyPoints) +
+                    (totalClassesExcused * excusedPoints) +
+                    (totalClassesMissed * absentPoints)) / totalClasses) * Double.parseDouble(attendanceAssignment.getAssignmentPoints()));
+            }
         } else {
 
-            return ((100 - entry.getPercentCourseMissed()) / 100) * Double.parseDouble(attendanceAssignment.getAssignmentPoints());
+            for (AttendanceSummaryModel model : summaryForSections) {
+                for (AttendanceSummaryModel.Entry entry : model.getEntries()) {
+                    if (entry.getSisUserId() != null && entry.getSisUserId().equals(student.getSisUserId())) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS134)